apps/api/v2/src/ee/event-types-private-links/services/private-links.service.ts

```
@@ -118,4 +137,26 @@ export class PrivateLinksService {
       throw new BadRequestException("Failed to delete private link");
     }
   }
+
+  private async resolveEventTypeSlug(
+    eventTypeId: number,
+    providedSlug?: string
+  ): Promise<string | undefined> {
+    if (providedSlug) return providedSlug;
+    const slug = await this.eventTypesRepository.getEventTypeSlugById(eventTypeId);
+    return slug ?? undefined;
+  }
+
+  private generateBookingUrl(hashedLink: string, orgSlug?: string, eventTypeSlug?: string): string {
+    if (orgSlug && eventTypeSlug) {
+      const origin = getOrgFullOrigin(orgSlug, { protocol: true }).replace(/\/$/, "");
+      return `${origin}/d/${hashedLink}/${eventTypeSlug}`;
+    }
+
+    const fallbackOrigin = process.env.NEXT_PUBLIC_WEBAPP_URL || "https://cal.com";
+    if (eventTypeSlug) {
+      return `${fallbackOrigin}/d/${hashedLink}/${eventTypeSlug}`;
+    }
+    return `${fallbackOrigin}/d/${hashedLink}`;
```

This is an invalid link, we should throw error instead of generating invalid link

For hashed links eventTypeSlug is required as they are always for an eventType

---

DevinAi implement feedback